src/LeverageZapRouterBase.sol

```diff
@@ -0,0 +1,231 @@
+import {ICdpManagerData} from "@ebtc/contracts/Interfaces/ICdpManager.sol";
+import {LeverageMacroBase} from "@ebtc/contracts/LeverageMacroBase.sol";
+import {IEbtcLeverageZapRouter} from "./interface/IEbtcLeverageZapRouter.sol";
+import {ZapRouterBase} from "./ZapRouterBase.sol";
+import {IStETH} from "./interface/IStETH.sol";
+import {IERC20} from "@ebtc/contracts/Dependencies/IERC20.sol";
+import "forge-std/console2.sol";
+
+abstract contract LeverageZapRouterBase is ZapRouterBase, LeverageMacroBase {
+    uint256 internal constant PRECISION = 1e18;
+
+    address internal immutable theOwner;
+    IPriceFeed internal immutable priceFeed;
+    address internal immutable dex;
+
+    constructor(
+        IEbtcLeverageZapRouter.DeploymentParams memory params
+    )
+        ZapRouterBase(IERC20(params.weth), IERC20(params.wstEth), IStETH(params.stEth))
+        LeverageMacroBase(
+            params.borrowerOperations,
+            params.activePool,
+            params.cdpManager,
+            params.ebtc,
+            params.stEth,
+            params.sortedCdps,
+            false // Do not sweep
+        )
+    {
+        theOwner = msg.sender;
+        priceFeed = IPriceFeed(params.priceFeed);
+        dex = params.dex;
+
+        // Infinite Approvals @TODO: do these stay at max for each token?
+        ebtcToken.approve(address(borrowerOperations), type(uint256).max);
+        stETH.approve(address(borrowerOperations), type(uint256).max);
+        stETH.approve(address(activePool), type(uint256).max);
+    }
+
+    function owner() public override returns (address) {
+        return theOwner;
+    }
+
+    function _sweep() private {
+        /**
+         * SWEEP TO CALLER *
+         */
+        // Safe unchecked because known tokens
+        uint256 ebtcBal = ebtcToken.balanceOf(address(this));
+        uint256 collateralBal = stETH.sharesOf(address(this));
+
+        if (ebtcBal > 0) {
+            ebtcToken.transfer(msg.sender, ebtcBal);
+        }
+
+        console2.log("collateralBal", collateralBal);
+
+        if (collateralBal > 0) {
+            stETH.transferShares(msg.sender, collateralBal);
+        }
+    }
+
+    function _debtToCollateral(uint256 _debt) public returns (uint256) {
+        uint256 price = priceFeed.fetchPrice();
+        return (_debt * PRECISION) / price;
+    }
+
+    function _adjustCdpOperation(
+        bytes32 _cdpId, 
+        FlashLoanType _flType,
+        uint256 _flAmount,
+        uint256 _marginIncrease,
+        AdjustCdpOperation memory _cdp,
+        uint256 newDebt,
+        uint256 newColl,
+        bytes calldata _exchangeData
+    ) internal {
+        LeverageMacroOperation memory op;
+
+        op.tokenToTransferIn = address(stETH);
+        op.amountToTransferIn = _marginIncrease;
+        op.operationType = OperationType.AdjustCdpOperation;
+        op.OperationData = abi.encode(_cdp);
+
+        if (_cdp._isDebtIncrease) {
```

Comment:
does the `_flType` parameter play a role here for `op.swapAfter`? 

For example, if `_flType` is `eBTC`, and with `_cdp._isDebtIncrease=true`, do we still need to swap after FL?